### PR TITLE
cfssl: Add version 1.5.0

### LINKED ISSUE #5104: [Request] Add cfssl - CloudFlare's PKI/TLS toolkit
Requesting https://github.com/cloudflare/cfssl

I might try to contribute this myself.

### PR DESCRIPTION
**Pull request by kennedy**
- Closes #5104

The `cfssl` is really a suite of utilities including `cfssl-bundle`, `cfssl-certinfo`, `cfssl-newkey`, `cfssl-scan`, `cfssljson`, `mkbundle`, `multirootca`. I've include them all in the `url`, `bin`, and `hash`

However, `autoupdate` only recognizes and updates the hash for the base app `cfssl`, but not the other utilities, even thought they are in the `autoupdate.hash` file

Do I:

1. Make individual app manifest for each utility, and have them listed as `depends/suggest` on `cfssl`?
2. Is there a special configuration I am missing for this use case?

### CODE REVIEW

**Review by Ash258 — COMMENTED** [review]
Remove autoupdate

File: bucket/cfssl.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{
+    "version": "1.5.0",
+    "description": "CloudFlare's PKI/TLS toolkit",
```
Comment (Ash258) [inline bucket/cfssl.json:3]: Completely saying-nothing description

File: bucket/cfssl.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    "homepage": "https://github.com/cloudflare/cfssl",
```
Comment (Ash258) [inline bucket/cfssl.json:4]: Use url from repository description

**Review by Ash258 — CHANGES_REQUESTED** [review]

File: bucket/cfssl.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    "license": "BSD-2-Clause",
+    "url": [
+        "https://github.com/cloudflare/cfssl/releases/download/v1.5.0/cfssl_1.5.0_windows_amd64.exe",
```
Comment (Ash258) [inline bucket/cfssl.json:7]: This is obviously 64bit only

**Review by Ash258 — CHANGES_REQUESTED** [review]
This has to fail on installation immediatelly on first shim.

**Comment by kennedy** [comment]
> Remove autoupdate

Is that the usual solution for situations like this?

File: bucket/cfssl.json (hunk 1 above)
Comment (kennedy) [reply bucket/cfssl.json:3]: I will add the official des

File: bucket/cfssl.json (hunk 3 above)
Comment (kennedy) [reply bucket/cfssl.json]: ill add the arch section

**Comment by Ash258** [comment]
Stop wasting github actions time on each line you change. Completely nonsense. Setup local environment instead

File: bucket/cfssl.json [hunk 4]
```diff
@@ -0,0 +1,41 @@
+{
+    "version": "1.5.0",
+    "description": "CFSSL is CloudFlare's PKI/TLS swiss army knife. It is both a command line tool and an HTTP API server for signing, verifying, and bundling TLS certificates",
+    "homepage": "https://cfssl.org/",
```
Comment (Ash258) [inline bucket/cfssl.json:4]:
```suggestion
    "description": "Command line tool and an HTTP API server for signing, verifying, and bundling TLS certificates.",
    "homepage": "https://cfssl.org",
```

File: bucket/cfssl.json [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+            ],
+            "bin": [
+                "cfssl.exe",
+                "cfssl-bundle.exe",
+                "cfssl-certinfo.exe",
+                "cfssl-newkey.exe",
+                "cfssl-scan.exe",
+                "cfssljson.exe",
+                "mkbundle.exe",
+                "multirootca.exe"
```
Comment (Ash258) [inline bucket/cfssl.json:34]:
```suggestion
```

File: bucket/cfssl.json [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+            ]
+        }
+    },
+    "checkver": {
```
Comment (Ash258) [inline bucket/cfssl.json:38]:
```suggestion
    "bin": [
        "cfssl.exe",
        "cfssl-bundle.exe",
        "cfssl-certinfo.exe",
        "cfssl-newkey.exe",
        "cfssl-scan.exe",
        "cfssljson.exe",
        "mkbundle.exe",
        "multirootca.exe"
    ],
    "checkver": {
```

File: bucket/cfssl.json [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+            ],
+            "hash": [
+                "fce669f20d9f25dce20e4a39742613434e16c444ef5745aa91e546223031dbd5",
+                "17282aba7a47205be7b937b541af1f33f75654afc66d3a54081a23a20273ad14",
+                "39cb0f739fec07ae8b16d3fd7e07929656748a1eb5b12e0632d1864282eb5d76",
+                "0e7371833942508e939beb051c09aa4112f3e4743460d549d23f8aace5ab0239",
+                "6fd4776e044489f1858ed33e7cfe09728396f1d01e5d5d5934b78c7ab5cddab9",
+                "aa3c8c84ac3c0e8ff69d798a01ea25fdd15934b92f7979c9c1cebca845446f7e",
+                "cea263e6e58aa20d924bc5f04aa67882a5a1660a2d05d473a192efdf8648b066"
+            ],
```
Comment (Ash258) [inline bucket/cfssl.json:25]:
```suggestion
            "url": [
                "https://github.com/cloudflare/cfssl/releases/download/v1.5.0/cfssl_1.5.0_windows_amd64.exe#/cfssl.exe",
                "https://github.com/cloudflare/cfssl/releases/download/v1.5.0/cfssljson_1.5.0_windows_amd64.exe#/cfssljson.exe",
                "https://github.com/cloudflare/cfssl/releases/download/v1.5.0/cfssl-bundle_1.5.0_windows_amd64.exe#/cfssl-bundle.exe",
                "https://github.com/cloudflare/cfssl/releases/download/v1.5.0/cfssl-certinfo_1.5.0_windows_amd64.exe#/cfssl-certinfo.exe",
                "https://github.com/cloudflare/cfssl/releases/download/v1.5.0/cfssl-newkey_1.5.0_windows_amd64.exe#/cfssl-newkey.exe",
                "https://github.com/cloudflare/cfssl/releases/download/v1.5.0/cfssl-scan_1.5.0_windows_amd64.exe#/cfssl-scan.exe",
                "https://github.com/cloudflare/cfssl/releases/download/v1.5.0/mkbundle_1.5.0_windows_amd64.exe#/mkbundle.exe",
                "https://github.com/cloudflare/cfssl/releases/download/v1.5.0/multirootca_1.5.0_windows_amd64.exe#/multirootca.exe"
            ],
            "hash": [
                "fce669f20d9f25dce20e4a39742613434e16c444ef5745aa91e546223031dbd5",
                "6fd4776e044489f1858ed33e7cfe09728396f1d01e5d5d5934b78c7ab5cddab9",
                "3ecc18c7e168d611d63f84d63778a8c629b4223718d02094e4e8124cfc18440a",
                "17282aba7a47205be7b937b541af1f33f75654afc66d3a54081a23a20273ad14",
                "39cb0f739fec07ae8b16d3fd7e07929656748a1eb5b12e0632d1864282eb5d76",
                "0e7371833942508e939beb051c09aa4112f3e4743460d549d23f8aace5ab0239",
                "aa3c8c84ac3c0e8ff69d798a01ea25fdd15934b92f7979c9c1cebca845446f7e",
                "cea263e6e58aa20d924bc5f04aa67882a5a1660a2d05d473a192efdf8648b066"
            ]
```

**Review by Ash258 — CHANGES_REQUESTED** [review]

**Review by kennedy — COMMENTED** [review]
I will add these changes

File: bucket/cfssl.json [hunk 8]
```diff
@@ -0,0 +1,41 @@
+{
+    "version": "1.5.0",
+    "description": "Command line tool and an HTTP API server for signing, verifying, and bundling TLS certificates.",
+    "homepage": "https://cfssl.org",
+    "license": "BSD-2-Clause",
+    "architecture": {
+        "64bit": {
+            "url": [
+                "https://github.com/cloudflare/cfssl/releases/download/v1.5.0/cfssl_1.5.0_windows_amd64.exe#/cfssl.exe",
```
Comment (Ash258) [inline bucket/cfssl.json:9]:
```suggestion
                "https://github.com/cloudflare/cfssl/releases/download/v1.5.0/cfssl_1.5.0_windows_amd64.exe#/cfssl.exe",
                "https://github.com/cloudflare/cfssl/releases/download/v1.5.0/cfssl-bundle_1.5.0_windows_amd64.exe#/cfssl-bundle.exe",
```

File: bucket/cfssl.json [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+                "https://github.com/cloudflare/cfssl/releases/download/v1.5.0/cfssl-certinfo_1.5.0_windows_amd64.exe#/cfssl-certinfo.exe",
+                "https://github.com/cloudflare/cfssl/releases/download/v1.5.0/cfssl-newkey_1.5.0_windows_amd64.exe#/cfssl-newkey.exe",
+                "https://github.com/cloudflare/cfssl/releases/download/v1.5.0/cfssl-scan_1.5.0_windows_amd64.exe#/cfssl-scan.exe",
+                "https://github.com/cloudflare/cfssl/releases/download/v1.5.0/cfssljson_1.5.0_windows_amd64.exe#/cfssljson.exe",
+                "https://github.com/cloudflare/cfssl/releases/download/v1.5.0/mkbundle_1.5.0_windows_amd64.exe#/mkbundle.exe",
+                "https://github.com/cloudflare/cfssl/releases/download/v1.5.0/multirootca_1.5.0_windows_amd64.exe#/multirootca.exe"
+            ],
+            "hash": [
+                "fce669f20d9f25dce20e4a39742613434e16c444ef5745aa91e546223031dbd5",
```
Comment (Ash258) [inline bucket/cfssl.json:18]:
```suggestion
                "fce669f20d9f25dce20e4a39742613434e16c444ef5745aa91e546223031dbd5",
                "3ecc18c7e168d611d63f84d63778a8c629b4223718d02094e4e8124cfc18440a",
```

**Review by Ash258 — CHANGES_REQUESTED** [review]

**Review by Ash258 — APPROVED** [review]

**Pull request merged**